interfaces/repo.go

```diff
@@ -939,16 +939,49 @@ func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string, policy
 			if slot.Interface != plug.Interface {
 				continue
 			}
+			iface := slot.Interface
 
 			// declaration based checks disallow
 			if !policyCheck(plug, slot) {
 				continue
 			}
 
-			if r.ifaces[plug.Interface].AutoConnect(plug, slot) {
+			if r.ifaces[iface].AutoConnect(plug, slot) {
 				candidates = append(candidates, slot)
 			}
 		}
 	}
 	return candidates
 }
+
+// AutoConnectCandidatePlugs finds and returns viable auto-connection candidates
+// for a given slot.
+func (r *Repository) AutoConnectCandidatePlugs(slotSnapName, slotName string, policyCheck func(*Plug, *Slot) bool) []*Plug {
```

Comment:
This looks very similar to AutoConnectCandidateSlots - is there a way to consolidate the common bits into a helper maybe?